js/components/ImportIndicatorsPopover.js

```diff
@@ -1,142 +1,271 @@
+    let handleUpload = (e) => {
+        let files = e.target.files;
+        let reader = new FileReader();
+        reader.readAsDataURL(files[0]);
+        reader.onload = (e) => {
+            api.uploadTemplate(e.target.result)
             .then(response => {
                 console.log("Reponse:", response);
-                alert("DONWLOAD TEMPLATE")
+                alert("UPLOAD TEMPLATE");
+                setViews(FEEDBACK);
             })
+        }
     }
-    handleUpload = () => {
+    // Triggers the file upload from the Upload button
+    let uploadClick = () => {
         console.log("Upload Clicked");
-        api.uploadTemplate()
-        .then(response => {
-            console.log("Reponse:", response);
-            alert("UPLOAD TEMPLATE")
-        })
+        document.getElementById("fileUpload").click();
     }
 
-    render() {
-        return (
-            <React.Fragment>
-                {(() => {
-                    switch(this.state.status) {
-                        // View for downloading and uploading the template
-                        case this.DOWNLOAD_UPLOAD:
-                            return (
-                                <div className="importIndicators-body">
-                                    <div className="import-text">
-                                        <ol>
-                                            <li>
-                                                {
-                                                    // # Translators: Instructions for users to download the template, open it in Excel, and then fill in indicators information.
-                                                    gettext("Download the template, open it in Excel, and enter indicators.")
-                                                }
-                                            </li>
-                                            <li>
-                                                {
-                                                    // # Translators: Instruction for users to upload their filled in template and then follow the instructions to complete the import process.
-                                                    gettext("Upload the template and follow instructions to complete the process.")
-                                                }
-                                            </li>
-                                        </ol>
-                                    </div>
-
-                                    <div className="import-buttons">
-                                        <button
-                                            type="button"
-                                            className="btn btn-sm btn-primary btn-download"
-                                            onClick={() => this.handleDownload()}
-                                            >
+    // Creates the default number of rows for the Tier Levels
+    chosenTier.map((tier, i) => {
```

Comment:
You could turn this into a function that is used to create the defaults for the state when the state is created, or something similar that more tightly couples the defaults with state creation.